src/lib.rs

```diff
@@ -172,19 +175,19 @@ pub enum Destination {
 /// routing messages.
 ///
 /// See the [crate documentation](index.html) for details.
-pub struct RoutingTable {
+pub struct RoutingTable<T> {
     /// The buckets, by bucket index. Each bucket is sorted by ascending distance from us.
-    buckets: Vec<Vec<XorName>>,
-    /// This nodes' own name.
-    our_name: XorName,
+    buckets: Vec<Vec<NodeInfo<T>>>,
+    /// This nodes' own info.
+    our_info: NodeInfo<T>,
```

Comment:
Had to replace `our_name` with `our_info`, because we return it from `closest_nodes_to`.